### PR TITLE
[WIP] AD713x driver update

### PR DESCRIPTION
**Pull request by adrimbarean**
Update the AD713X driver to the new silicon revision.
Changes this patch:
 - update the register map;
 - update implementation to new register names;
 - update coding style to a more Linux friendly format.

### CODE REVIEW

File: ad713x-fmcz/ad713x.h [hunk 1]
```diff
@@ -60,79 +60,79 @@
 #define AD713X_REG_PRODUCT_ID_LSB		0x04
 #define AD713X_REG_PRODUCT_ID_MSB		0x05
 #define AD713X_REG_CHIP_GRADE			0x06
```
Comment (mirceacaprioru) [inline ad713x-fmcz/ad713x.h]: are these modifications reflected in a datasheet ? maybe add a link to it in the commit message

File: ad713x-fmcz/ad713x.h [hunk 2]
```diff
@@ -643,45 +643,51 @@ typedef enum ad713x_supported_dev_ids {
 	ID_AD7136
 } ad713x_supported_dev_ids;
 
-typedef enum {
-	LOW_POWER = 0x01,
+typedef enum ad713x_power_mode {
```
Comment (mirceacaprioru) [inline ad713x-fmcz/ad713x.h]: i don't know if we should use this intermediary step. This part is in its early stages so we can't just talk yet about backward compatibility. We should do it the right way from now on.

File: ad713x-fmcz/ad713x.h [hunk 3]
```diff
@@ -643,45 +643,51 @@ typedef enum ad713x_supported_dev_ids {
...
 
-typedef enum {
+typedef enum ad713x_channels {
 	CH0,
 	CH1,
 	CH2,
 	CH3
 } ad713x_channels;
 
+enum ad717x_mpc_clkdel {
```
Comment (mirceacaprioru) [inline ad713x-fmcz/ad713x.h]: here you did not use typedef so this is a first sign of inconsistency.

File: ad713x-fmcz/ad713x.c [hunk 4]
```diff
@@ -151,14 +151,18 @@ int32_t ad713x_set_power_mode(ad713x_dev *dev,
 int32_t ad713x_dout0_enable(ad713x_dev *dev,
 			    bool dout0_en)
 {
-	int32_t ret;
+	/* Keep function for backwards compatibility.
```
Comment (mirceacaprioru) [inline ad713x-fmcz/ad713x.c]: again i am not convinced about the backwards compatibility.

File: ad713x-fmcz/ad713x.c [hunk 5]
```diff
@@ -239,15 +245,55 @@ int32_t ad713x_mag_phase_clk_delay(ad713x_dev *dev,
...
 {
 	int32_t ret;
+	int8_t i;
 
-	ret = ad713x_spi_write_mask(dev,
-				    AD713X_REG_DEVICE_CONFIG5,
-				    AD713X_DEV_CONFIG5_MPC_CLKDEL_EN_MSK,
-				    AD713X_DEV_CONFIG4_MPC_CLKDEL_EN_MODE(clk_delay_en));
+	if(clk_delay_en) {
+		for(i = CH3; i >= 0; i--) {
```
Comment (mirceacaprioru) [inline ad713x-fmcz/ad713x.c]: use just one for and a aux variable for AD713X_MPC_CLKDEL_EN_CH_MODE. 

Then you do this:
if(clk_delay_en)
    temp = DELAY_2_CLOCKS;
else 
    temp = DELAY_NONE;
for(i = CH3; i >= 0; i--) {
			ret = ad713x_spi_write_mask(dev,
						    AD713X_REG_MPC_CONFIG,
						    AD713X_MPC_CLKDEL_EN_CH_MSK(i),
						    AD713X_MPC_CLKDEL_EN_CH_MODE(temp, i));
			if(ret != 0)
				break;
}

File: ad713x-fmcz/ad713x.c [hunk 6]
```diff
@@ -192,9 +192,9 @@ int32_t ad713x_set_out_data_frame(ad713x_dev *dev,
 		if((adc_data_len == ad713x_output_data_frame[id][i][0]) &&
 		    (crc_header == ad713x_output_data_frame[id][i][1])) {
 			ret = ad713x_spi_write_mask(dev,
```
Comment (mirceacaprioru) [inline ad713x-fmcz/ad713x.c]: return directly ?

**Comment by adrimbarean** [comment]
V0 -> V1:
 - scrap backwards compatibility since it's a new driver;
 - improve driver consistency;
 - refactor code to be on par with new standards (return dirrectly when possible; use updated API templates for initialization functions, etc.);
 - remove unnecessary 'for'.

**Comment by adrimbarean** [comment]
V1 -> V2:
 - fix compilation errors.

**Comment by adrimbarean** [comment]
V2 -> V3:
 - update and improve the initialization routine after testing with hardware.

**Comment by adrimbarean** [comment]
V3 -> V4:
 - fix compile mistakes made when refactoring code.

**Pull request closed without merging**